src/ConnectionGraphicsObject.cpp

```diff
@@ -212,17 +241,39 @@ mouseReleaseEvent(QGraphicsSceneMouseEvent* event)
 
   auto node = locateNodeAt(event->scenePos(), _scene,
                            _scene.views()[0]->transform());
+  
+  if (!node) {
+    
+    if (state().requiresPort()) {
+      Q_ASSERT(this == _scene._temporaryConn);
+      // remove this from the scene
+      delete _scene._temporaryConn;
+      _scene._temporaryConn = nullptr;
+    }
+    
+    return;
+  }
+  
+  if (!state().requiresPort()) {
+    return;
+  }
 
-  NodeConnectionInteraction interaction(*node, _connection, _scene);
+  NodeConnectionInteraction interaction(node->index(), *this);
 
   if (node && interaction.tryConnect())
   {
     node->resetReactionToConnection();
+    Q_ASSERT(this == _scene._temporaryConn);
+
+    delete _scene._temporaryConn;
+    _scene._temporaryConn = nullptr;
   }
-  else if (_connection.connectionState().requiresPort())
+  else if (state().requiresPort())
```

Comment:
Can this occur? In line 257 we return early for the opposite case. Did the state change in between? Also, see my comment above.